README.md

```diff
@@ -1,2 +1,422 @@
+
+Types that are declared outside of your project can be instantiated by SafeDI if there is an extension on the type decorated with the [`@ExternalInstantiable` macro](Sources/SafeDI/PropertyDecoration/ExternalInstantiable.swift). Extensions decorated with this macro define how to instantiate the extended type via a `public func instantiate(…) -> ExtendedType` method. This `instantiate(…)` method can receive dependencies instantiated or forwarded by objects further up the dependency tree by declaring these dependencies as method arguments.
+
+#### Example
+
+Here is a sample `SecurePersistentStorage` protocol whose concrete type is defined in a third-party dependency, and therefore is `@ExternalInstantiable`:
+
+```swift
+import Foundation
+import SafeDI
+import Valet // A Keychain wrapper that can be used to implement secure, persistent storage. github.com/square/valet
+
+/// A protocol defining how to interact with secure, persistent storage.
+/// It is not necessary to utilize protocols with SafeDI, but since
+/// protocol-driven development aids both testability and dependency
+/// inversion, our examples show how protocols can be used with SafeDI.
+protocol SecurePersistentStorage {
+    subscript<CodableType: Codable>(_ key: String, ofType type: CodableType.Type) -> CodableType? { get }
+    subscript<CodableType: Codable>(_ key: String) -> CodableType? { get set }
+}
+
+/// A default implementation of `SecurePersistentStorage` that can fulfill
+/// `@Instantiated` properties of type `Valet` or `SecurePersistentStorage`.
+@ExternalInstantiable(fulfillingAdditionalTypes: [SecurePersistentStorage.self])
+extension Valet: SecurePersistentStorage {
+
+    /// A public initializer defines how SafeDI can instantiate a Valet object.
+    public static func instantiate() -> Valet {
+        Valet.valet(
+            with: Identifier(nonEmpty: "SafeDIExample")!,
+            accessibility: .afterFirstUnlock
+        )
+    }
+
+    subscript<CodableType: Codable>(_ key: String, ofType type: CodableType.Type) -> CodableType? {
+        guard let data = try? object(forKey: key) else { return nil }
+        return try? JSONDecoder().decode(type, from: data)
+    }
+
+    subscript<CodableType: Codable>(key: String) -> CodableType? {
+        get {
+            self[key, ofType: CodableType.self]
+        }
+        set {
+            if let newValue, let data = try? JSONEncoder().encode(newValue) {
+                try? setObject(data, forKey: key)
+            } else {
+                try? removeObject(forKey: key)
+            }
+        }
+    }
+}
+```
+
+### @Instantiated
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Instantiated` macro](Sources/SafeDI/PropertyDecoration/Instantiated.swift) are instantiated when its enclosing type is instantiated. `@Instantiated`-decorated properties are available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
```

Comment:
```suggestion
Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Instantiated`](Sources/SafeDI/PropertyDecoration/Instantiated.swift) are instantiated when its enclosing type is instantiated. `@Instantiated`-decorated properties are available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
```
It has been defined earlier as a Macro, no need for repeating.

Reply:
😅 can you tell which sections I wrote first? yeah I can simplify here.